its/ruling/src/integrationTest/java/org/sonarsource/iac/IacRulingTest.java

```diff
@@ -219,22 +220,20 @@ private static void activateJvmFrameworkConfigJavaRulesProfile() throws IOExcept
       .append("  <language>java</language>\n")
       .append("  <rules>\n");
 
+    Set<String> keys = new HashSet<>();
     SpringConfigCheckList.checks().stream()
       .map(IacRulingTest::annotatedRuleKey)
       .filter(Objects::nonNull)
-      .forEach(ruleKey -> sb.append("    <rule>\n")
-        .append("      <repositoryKey>java</repositoryKey>\n")
-        .append("      <key>").append(ruleKey).append("</key>\n")
-        .append("      <priority>INFO</priority>\n")
-        .append("    </rule>\n"));
+      .forEach(keys::add);
     MicronautConfigCheckList.checks().stream()
       .map(IacRulingTest::annotatedRuleKey)
       .filter(Objects::nonNull)
-      .forEach(ruleKey -> sb.append("    <rule>\n")
-        .append("      <repositoryKey>java</repositoryKey>\n")
-        .append("      <key>").append(ruleKey).append("</key>\n")
-        .append("      <priority>INFO</priority>\n")
-        .append("    </rule>\n"));
+      .forEach(keys::add);
+    keys.forEach(ruleKey -> sb.append("    <rule>\n")
+      .append("      <repositoryKey>java</repositoryKey>\n")
+      .append("      <key>").append(ruleKey).append("</key>\n")
+      .append("      <priority>INFO</priority>\n")
+      .append("    </rule>\n"));
```

Comment:
```suggestion
Stream.concat(
        SpringConfigCheckList.checks().stream(),
        MicronautConfigCheckList.checks().stream())
      .map(IacRulingTest::annotatedRuleKey)
      .filter(Objects::nonNull)
      .distinct() // Fail to startup with duplicated keys
      .forEach(ruleKey -> sb.append("    <rule>\n")
        .append("      <repositoryKey>java</repositoryKey>\n")
        .append("      <key>").append(ruleKey).append("</key>\n")
        .append("      <priority>INFO</priority>\n")
        .append("    </rule>\n"));
```

Going a step further with refactoring this part, explicitly removing duplicated keys with `Stream::distinct` and sharing the whole logic between the check lists.